TrafficCapture/coreUtilities/src/main/java/org/opensearch/migrations/tracing/RootOtelContext.java

```diff
@@ -0,0 +1,139 @@
+import lombok.Setter;
+import org.opensearch.migrations.Utils;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+public class RootOtelContext implements IRootOtelContext {
+    private final OpenTelemetry openTelemetryImpl;
+    private final String scopeName;
+    @Getter
+    @Setter
+    Exception observedExceptionToIncludeInMetrics;
+
+    public static OpenTelemetry initializeOpenTelemetryForCollector(@NonNull String collectorEndpoint,
+                                                                    @NonNull String serviceName) {
+        var serviceResource = Resource.getDefault().toBuilder()
+                .put(ResourceAttributes.SERVICE_NAME, serviceName)
+                .build();
+
+        final var spanProcessor = BatchSpanProcessor.builder(OtlpGrpcSpanExporter.builder()
+                        .setEndpoint(collectorEndpoint)
+                        .setTimeout(2, TimeUnit.SECONDS)
+                        .build())
+                .setScheduleDelay(100, TimeUnit.MILLISECONDS)
+                .build();
+        final var metricReader = PeriodicMetricReader.builder(OtlpGrpcMetricExporter.builder()
+                        .setEndpoint(collectorEndpoint)
+                        // see https://opentelemetry.io/docs/specs/otel/metrics/sdk_exporters/prometheus/
+                        // "A Prometheus Exporter MUST only support Cumulative Temporality."
+                        //.setAggregationTemporalitySelector(AggregationTemporalitySelector.deltaPreferred())
+                        .build())
+                .setInterval(Duration.ofMillis(1000))
+                .build();
+        final var logProcessor = BatchLogRecordProcessor.builder(OtlpGrpcLogRecordExporter.builder()
+                        .setEndpoint(collectorEndpoint)
+                        .build())
+                .build();
+
+        var openTelemetrySdk = OpenTelemetrySdk.builder()
+                .setTracerProvider(SdkTracerProvider.builder().setResource(serviceResource)
+                        .addSpanProcessor(spanProcessor).build())
+                .setMeterProvider(SdkMeterProvider.builder().setResource(serviceResource)
+                        .registerMetricReader(metricReader).build())
+                .setLoggerProvider(SdkLoggerProvider.builder().setResource(serviceResource)
+                        .addLogRecordProcessor(logProcessor).build())
+                .build();
+
+        // Add hook to close SDK, which flushes logs
+        Runtime.getRuntime().addShutdownHook(new Thread(openTelemetrySdk::close));
+        return openTelemetrySdk;
+    }
+
+    public static OpenTelemetry initializeOpenTelemetry(String collectorEndpoint, String serviceName) {
+        return Optional.ofNullable(collectorEndpoint)
+                .map(endpoint -> initializeOpenTelemetryForCollector(endpoint, serviceName))
+                .orElse(OpenTelemetrySdk.builder().build());
```

Comment:
Do we intend to ignore serviceName when a collectorEndpoint is not passed

Reply:
I've added some documentation, a warning, and have changed the name to 'initializeOpenTelemetryWithCollectorOrAsNoop'.